apiary_v2.apib

```diff
@@ -907,3 +907,363 @@ values for all roster fields in the template roster design is provided:
+              "original_request": "<the original request that was sent, which resulted in this error>"
+          }
+
++ Response 405 (application/json)
+
+    + Body
+
+          {
+              "code": "method_not_allowed",
+              "message": "Method used is not allowed",
+              "original_request": "<the original request that was sent, which resulted in this error>"
+          }
+
++ Response 406 (application/json)
+
+    + Body
+
+          {
+              "code": "not_acceptable",
+              "message": "Could not satisfy the request Accept header.",
+              "original_request": "<the original request that was sent, which resulted in this error>"
+          }
+
++ Response 415 (application/json)
+
+    + Body
+
+          {
+              "code": "unsupported_media_type",
+              "message": "Unsupported media type in request.",
+              "original_request": "Not able to parse the request: `Unsupported media type \"type sent\" in request.`"
+          }
+
+
+## Retrieve Order [/v2/orders/{order_id}/]
+## Retrieve an existing order [GET]
+
+After an order has been created, this endpoint can be used to retrieve order data for example to check the order state.
+
++ Parameters
+    + order_id (string) - ID of the Order in the form of a uuid
+
++ Request Retrieve Order
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 201 (application/json)
+
+    + Body
+
+          {}
+
++ Response 400 (application/json)
+    
+    + Body
+    
+          {
+             "code": "bad_request",
+             "message": "Request payload is not valid.",
+          }
+
++ Response 401 (application/json)
+
+    + Body
+
+          {
+              "code": "authentication_failed",
+              "message": "Incorrect authentication credentials.",
+          }
+
++ Response 403 (application/json)
+
+    + Body
+
+          {
+              "code": "permission_denied",
+              "message": "You do not have permission to perform this action.",
+          }
+
++ Response 405 (application/json)
+
+    + Body
+
+          {
+              "code": "method_not_allowed",
+              "message": "Method used is not allowed",
+          }
+
++ Response 406 (application/json)
+
+    + Body
+
+          {
+              "code": "not_acceptable",
+              "message": "Could not satisfy the request Accept header.",
+          }
+
++ Response 415 (application/json)
+
+    + Body
+
+          {
+              "code": "unsupported_media_type",
+              "message": "Unsupported media type in request.",
+          }
+
+## Retrieve Shipping Address [/v2/orders/{order_id}/shipping_address]
+## Retrieve shipping address for existing order[GET]
+
+When an order has progressed to the `ready_to_ship` state, this endpoint can be used to retrieve the shipping address 
+for the order. For orders not in this state, this endpoint will respond with an HTTP 404.
```

Comment:
Not sure if 404 it's a good idea or misleading. I will think 404 it's the order does not exist. Maybe there's a more appropriate code or even `200` and payload with message if it's not ready yet?

Reply:
That sounds better than 404.